File: src/test/java/eotree/data/TestEORegexData.java
package eotree.data;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

/**
 * EO regex data tests.
 */
public class TestEORegexData {

    private static final String regexSample = "/[a-z]+.+/m";

    @Test
    public void testGenerateEOZeroIndent() {
        var f = new EORegexData(regexSample);

        assertEquals(f.generateEO(0), regexSample);
    }

    @Test
    public void testGenerateEONonZeroIndent() {
        var f = new EORegexData(regexSample);

        assertEquals(f.generateEO(1), regexSample);
    }
}

File: src/test/kotlin/eotree/data/TestEORegexData.kt
package eotree.data

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.polystat.j2eo.eotree.data.EORegexData

/**
 * EO regex data tests.
 */
class TestEORegexData {
    @Test
    fun testGenerateEOZeroIndent() {
        val f = EORegexData(REGEX_SAMPLE)
        Assertions.assertEquals(f.generateEO(0), REGEX_SAMPLE)
    }

    @Test
    fun testGenerateEONonZeroIndent() {
        val f = EORegexData(REGEX_SAMPLE)
        Assertions.assertEquals(f.generateEO(1), REGEX_SAMPLE)
    }

    companion object {
        private const val REGEX_SAMPLE = "/[a-z]+.+/m"
    }
}
